packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1419,6 +1422,24 @@ abstract class BuildableElement extends Element {
     markNeedsBuild();
   }
 
+  bool _hadDependenciesWhenDeactivated = false;
```

Comment:
Looks like the rendering library just doesn't check if things have been inserted twice. I could do that in widgets too if you like. It's probably fine.